### PR TITLE
[PDE-4884] Automated tag and release on merge to main for kotlin

### PR DESCRIPTION
**Pull request by econte-sprig**
I cannot verify 100% that this works on my forked repo, due to not having the secrets to publish the android release (and not wanting to genuinely publish an android release to test it anyway) However I can verify it gets to the final step of publishing/building the android release and the inputs look correct to me.

image

### CODE REVIEW

File: .github/workflows/trigger_on_push_main.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    needs: get-tag
+    uses: UserLeap/common-workflows/.github/workflows/_generic_create_release_tag.yml@main
+    with:
+      tag: ${{ needs.get-tag.outputs.tag }}
+      release_url: https://api.github.com/repos/UserLeap/analytics-kotlin-sprig-test/releases
+    secrets: inherit
+  android-release:
+    if: ${{ needs.create-release-and-tag.outputs.new-tag == 'true' }}
+    needs: [create-release-and-tag,get-version]
+    uses: UserLeap/analytics-kotlin-sprig-test/.github/workflows/android_release.yml@main
```
Comment (gongstr) [inline .github/workflows/trigger_on_push_main.yml]: remove `-test`

File: gradle.properties [hunk 2]
```diff
@@ -18,4 +18,6 @@ android.useAndroidX=true
 # Automatically convert third-party libraries to use AndroidX
 android.enableJetifier=true
 # Kotlin code style for this project: "official" or "obsolete":
-kotlin.code.style=official
\ No newline at end of file
+kotlin.code.style=official
+
+VERSION=1.1.4
```
Comment (gongstr) [inline gradle.properties]: let's keep track of this version in a separate `version.properties` file. `gradle.properties` should be used for gradle related build properties.

File: .github/workflows/trigger_on_push_main.yml (hunk 1 above)
Comment (econte-sprig) [reply .github/workflows/trigger_on_push_main.yml]: good catch

File: gradle.properties (hunk 2 above)
Comment (econte-sprig) [reply gradle.properties]: fair enough

**Review by gongstr — APPROVED** [review]

**Pull request merged**